walnut/converter/converter.go

```diff
@@ -139,19 +144,50 @@ func TaskToYAML(extractedTasks []*task.Class) (err error) {
 	}
 
 	for _, SingleTask := range extractedTasks {
-		YAMLData, err := yaml.Marshal(SingleTask)
+		YAMLData, err := yaml.Marshal(&SingleTask)
 		if err != nil {
 			return fmt.Errorf("marshal failed: %v", err)
 		}
 		fileName := "tasks/" + SingleTask.Identifier.Name + ".yaml"
 		f, err := os.Create(fileName)
-		defer f.Close()
 
 		// Write marshaled YAML to file
 		err = ioutil.WriteFile(fileName, YAMLData, 0644)
 		if err != nil {
 			return fmt.Errorf("creating file failed: %v", err)
 		}
+		f.Close()
 	}
 	return
 }
+
+func RoleToYAML(input workflow.Role) (err error) {
+	// Check if "tasks" directory exists. If not, create it
+	_, err = os.Stat("workflows")
+	if os.IsNotExist(err) {
+		// FIXME: use absolute paths
+		errDir := os.Mkdir("workflows", 0755)
+		if errDir != nil {
+			return fmt.Errorf("create dir failed: %v", err)
+		}
+	}
+
+	yamlDATA, err := workflow.RoleToYAML(input)
+	if err != nil {
+		return fmt.Errorf("error converting role to YAML: %v", err)
+	}
+
+	var fileName = "workflows/"
+	if input.GetName() == "" {
```

Comment:
I think we should error out if there's no input name. How could this ever happen?